### PR TITLE
configfiles: don't save negative start values

### LINKED ISSUE #10346: [minor cosmetic] audio only can rewind to "-00"
### Important Information

I don't think this matters (expected to be universal)
- mpv version 0.32.0
- Debian 11/64bits
- Debian repos

### Reproduction steps

just run any audio file, or video file with --no-video. Pause, then rewind to the beginning with the arrow key. It will show "-00" (terminal and gui).

### Expected behavior

It should have shown "00"

### Actual behavior

It shows "-00"

### Other

Yes, this is really minor. I couldn't find it in the issues, so even if already fixed, it will be documented.

### PR DESCRIPTION
**Pull request by guidocella**
When you seek to the beginning of paused audio without video, the pts becomes negative, and if you then quit-watch-later, when mpv restarts playback it seeks to the end of the file and immediately advances to the next playlist entry. Fix this by making the pts 0 in this case.

Partial fix of #10346.

### CODE REVIEW

File: player/configfiles.c [hunk 1]
```diff
@@ -337,6 +337,15 @@ void mp_write_watch_later_conf(struct MPContext *mpctx)
...
     double pos = get_current_time(mpctx);
 
+    // Workaround pts being negative when pausing and seeking to the beginning
+    // of audio without video. This would save a negative start which would make
+    // mpv seek to the end and immediately advance to the next playlist entry.
+    // We can't make playing_audio_pts() always return positive values because
+    // it increases jitter with display-resample. ao_get_delay() should be fixed
+    // instead, as tracked in #12322.
+    if (pos < 0)
+        pos = 0;
```
Comment (kasper93) [inline player/configfiles.c:347]: This looks like hacky workaround for another problem. Also two lines below there is check for `MP_NOPTS_VALUE` which this change will break.

Comment (Dudemanguy) [reply player/configfiles.c:347]: Another possible approach would be to ensure that `get_current_time` never returns negative. I'm not sure if that has any implications, but I would be surprised if any current usage of that expects negative numbers ever.

Comment (kasper93) [reply player/configfiles.c:347]: Actually I don't understand why `playing_audio_pts` should be allowed to return negative values.

Comment (Dudemanguy) [reply player/configfiles.c:347]: Yeah, another possibility would be to have a separate function that does the delay calculation and stuff. I did remove the speed adjustment from it a while ago from it but we could go further.

Comment (guidocella) [reply player/configfiles.c:354]: Moved it below for now.

**Comment by Dudemanguy** [comment]
Superseded by https://github.com/mpv-player/mpv/pull/14227.

**Pull request closed without merging**